data/src/main/java/com/microsoft/azure/kusto/data/ClientImpl.java

```diff
@@ -416,6 +417,11 @@ private long determineTimeout(ClientRequestProperties properties, CommandType co
             throw new DataClientException(clusterUrl, "Failed to parse timeout from ClientRequestProperties");
         }
 
+        Object skipBoolean = properties == null ? null : properties.getOption(ClientRequestProperties.OPTION_SERVER_TIMEOUT);
```

Comment:
The way we figure out what timeouts to apply seem convoluted.
Maybe consolidate everything to this API and put it in 415 instead of trying to figure out what 415 really meant.

Reply:
Not sure I understand - 415 is irrelevant to this line. 

We don't check the result of 415 here, we independently check the header

Reply:
Why though?